### PR TITLE
Refactor unit tests

### PR DESCRIPTION
**Pull request by lukewrites**
- [x] Refactor unit tests to use `assertFalse`, `assertIsInstance`, and `assertEqual` as appropriate so we have more descriptive test failure messages.
- [x] fix tests that use mocks and add them to CI

We have some strange results right now. Three tests, `RerunFailedTestsTest` `LogicalTest` `ExpressionTest` get counted as failures when run by python's unittest library. However, stepping through the tests with the debugger or running them with pytest shows them as passing. I'm inclined to believe that they are in fact passing and that this is being caused by a python bug or unexpected behavior. The unittest package has changed between 3.4 and 3.7, which _might_ explain the changes?

### CODE REVIEW

**Comment by lukewrites** [comment]
```
rojects\connector-plugin-sdk\tdvt\test>python tdvt_test.py
Using root dir
Using test dir tool_test
..........................F.....F......F...........FFF............
======================================================================
FAIL: test_expression_tests (__main__.ExpressionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 134, in test_expression_tests
    self.check_results(all_test_results, 2)
  File "tdvt_test.py", line 116, in check_results
    self.assertTrue(len(test_results) == total_test_count, "Did not run the right number of tests. Ran {} rather than {}.".format(len(test_results), total_test_count))
AssertionError: False is not true : Did not run the right number of tests. Ran 0 rather than 2.

======================================================================
FAIL: test_logical_tests (__main__.LogicalTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 161, in test_logical_tests
    self.check_results(all_test_results, 1)
  File "tdvt_test.py", line 116, in check_results
    self.assertTrue(len(test_results) == total_test_count, "Did not run the right number of tests. Ran {} rather than {}.".format(len(test_results), total_test_count))
AssertionError: False is not true : Did not run the right number of tests. Ran 0 rather than 1.

======================================================================
FAIL: test_failed_test_output (__main__.ReRunFailedTestsTest)
Make sure TDVT writes the correct output file for rerunning failed tests.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 197, in test_failed_test_output
    self.check_results(all_test_results, 1, False)
  File "tdvt_test.py", line 116, in check_results
    self.assertTrue(len(test_results) == total_test_count, "Did not run the right number of tests. Ran {} rather than {}.".format(len(test_results), total_test_count))
AssertionError: False is not true : Did not run the right number of tests. Ran 0 rather than 1.

======================================================================
FAIL: test_results_missing_tuple (__main__.ResultsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 800, in test_results_missing_tuple
    self.assertIsInstance(mock_batch.results[test_file].error_status, TestErrorResults)
AssertionError: <tdvt.test_results.TestErrorMissingActual object at 0x04555EB0> is not an instance of <class 'tdvt.test_results.TestErrorResults'>

======================================================================
FAIL: test_results_passed (__main__.ResultsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 777, in test_results_passed
    self.assertTrue(mock_batch.results[test_file].all_passed())
AssertionError: False is not true

======================================================================
FAIL: test_results_wrong_value (__main__.ResultsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tdvt_test.py", line 820, in test_results_wrong_value
    self.assertIsInstance(mock_batch.results[test_file].error_status, TestErrorResults)
AssertionError: <tdvt.test_results.TestErrorMissingActual object at 0x04555A10> is not an instance of <class 'tdvt.test_results.TestErrorResults'>

----------------------------------------------------------------------
Ran 63 tests in 177.949s

FAILED (failures=6)
```

File: tdvt/test/tdvt_test.py [hunk 1]
```diff
@@ -734,8 +741,22 @@ def test_print_configuration_with_no_dsname(self):
             datasource_list.print_configurations(MockTestRegistry, None, None)
             self.assertIn(correct_out, captured_output.getvalue())
 
+
```
Comment (lukewrites) [inline tdvt/test/tdvt_test.py]: Moved this from below to allow for type hints.

File: tdvt/tdvt/config/tdvt/tdvt.ini [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [DEFAULT]
-TAB_CLI_EXE_X64 = C:/Program Files/Tableau/Tableau 2019.2/bin/tabquerytool.exe
-TAB_CLI_EXE_MAC = /Applications/Tableau 2019.2.app/Contents/MacOS/tabquerytool
\ No newline at end of file
+TAB_CLI_EXE_X64 = C:/Program Files/Tableau/Tableau 2019.3/bin/tabquerytool.exe
```
Comment (lukewrites) [inline tdvt/tdvt/config/tdvt/tdvt.ini]: should we change all of these references to 2019.4?

Comment (lriggs) [reply tdvt/tdvt/config/tdvt/tdvt.ini]: Maybe we could set them to something non-specific and then rely on setting the env variable when running the tests locally.

**Comment by lukewrites** [comment]
> 
> 
> Are the new actual and diff files needed for a test?

@lriggs whoops I shouldn't have added the diff files, I'll remove them.

The actuals weren't getting generated on my machine and it didn't seem like they were generated by the original PR either when I checked out to it and ran the tests. the path that's being fed to `MockBatchQueueWork.test_list_path` looks wrong. Do we want these tests to generate the actual files?

**Comment by lriggs** [comment]
Yeah I think having them checked in was the intent.

**Comment by lriggs** [comment]
I was wondering why these passed locally for me and it turns out I have the actual file but since the .gitignore file filters out actual.setup*.txt I never added it to my commit.

File: tdvt/tdvt/config/tdvt/tdvt.ini (hunk 2 above)
Comment (lukewrites) [reply tdvt/tdvt/config/tdvt/tdvt.ini]: Good idea, I reverted the changes.

**Pull request merged**